pypechain/core/utilities.py

```diff
@@ -40,10 +37,10 @@ def tuple_to_dataclass(cls: type[T], structs: dict[str, Any], tuple_data: Any |
     field_values = {}
 
     for (field_name, field_type), value in zip(field_types.items(), tuple_data):
-        field_type = structs.get(field_type, field_type)
+        field_type = structs.get(field_type, field_type)  # type: ignore
```

Comment:
Can you use type narrowing before this line to not have to do `type: ignore`?

(not a big deal, just commenting since this is a typing library. So feel free to resolve the comment without fixing it. )